Source/Blazorise/Components/Addon/Addons.razor.cs

```diff
@@ -67,7 +65,7 @@ protected override void Dispose( bool disposing )
     protected override void BuildClasses( ClassBuilder builder )
     {
         builder.Append( ClassProvider.Addons() );
-        builder.Append( ClassProvider.AddonsSize( ThemeSize ), ThemeSize != Blazorise.Size.Default );
+        builder.Append( ClassProvider.AddonsSize( ThemeSize ) );
```

Comment:
Only `ThemeSize` here?

Reply:
Ah, it's a calculated Size. Why is it not named `Size`?

Reply:
Because we already have the Size parameter.

Reply:
You got me thinking. We could just use a property-getter instead, eg. 

```cs
[Parameter]
public Size? Size
{
    get => size.GetValueOrDefault( Theme?.InputOptions?.Size ?? Blazorise.Size.Default );
    set
    {
        size = value;

        DirtyClasses();
    }
}
```

Reply:
Looks good I guess?

Reply:
But, this will then go against the Auto parameters, #4414 ?

Reply:
I mean, they are guidelines, they are there to try and stop people from doing unsupported stuff or causing unitended side effects.
I believe the backed field is fine.

However I believe `DirtyClasses` may cause a re-render tough, and maybe it would be avoided if it was called on the `SetParametersAsync` instead? 
We should test this in the future. I was just reading this and I Find it interesting, we do this on Datagrid a lot for example. 

We set SelectedRow and SelectedRows and also call their Changed counterparts, and maybe this is a big mistake and what was causing the 2x SetParametersAsync + 3x Renders on the Datagrid improvements PR I pushed a few days ago.
Maybe when there's a `Value` Parameter which also has a `ValueChanged` Parameter, it makes sense to just call the `ValueChanged` with the new value and not mutate the Parameter. Makes sense?
![image](https://user-images.githubusercontent.com/22283161/209241662-3ff51da1-8f2f-4722-b80a-71e148a80d3e.png)

Reply:
> I mean, they are guidelines, they are there to try and stop people from doing unsupported stuff or causing unitended side effects. I believe the backed field is fine.
> 
> However I believe `DirtyClasses` may cause a re-render tough, and maybe it would be avoided if it was called on the `SetParametersAsync` instead? We should test this in the future. I was just reading this and I Find it interesting, we do this on Datagrid a lot for example.
> 
> We set SelectedRow and SelectedRows and also call their Changed counterparts, and maybe this is a big mistake and what was causing the 2x SetParametersAsync + 3x Renders on the Datagrid improvements PR I pushed a few days ago. Maybe when there's a `Value` Parameter which also has a `ValueChanged` Parameter, it makes sense to just call the `ValueChanged` with the new value and not mutate the Parameter. Makes sense? PS: I think Autocomplete has the same problem. ![image](https://user-images.githubusercontent.com/22283161/209241662-3ff51da1-8f2f-4722-b80a-71e148a80d3e.png)

I don't think removing the Value update would work. As far as I know, and remember from the past that kind of change would break the component lifecycle. But as always, it is best to test and see. We can do that as part of a larger refactor later on.

Reply:
Better to leave it as `ThemeSize`. Otherwise, we would have to check everywhere if Size `is not null` and is not `Size.Default`. And there is a lot of places.